src/main/java/com/cburch/logisim/gui/main/TickCounter.java

```diff
@@ -13,172 +13,107 @@
 
 import com.cburch.logisim.circuit.Simulator;
 
-class TickCounter implements Simulator.Listener {
-  private static final int QUEUE_LENGTH = 1000;
-
-  private final long[] queueTimes;
-  private final double[] queueRates;
-  private int queueStart;
-  private int queueSize;
-  private double tickFrequency;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+public class TickCounter implements Simulator.Listener {
+  private final DecimalFormat formatter;
+  private Simulator simulator;
+  private long tickCount = 0;
+  private long startTime;
+  private boolean useKiloHertz = false;
+  private double previousFrequency = 0.0;
+  private long elapsedTimeSinceLastUnitUpdate = 0;
+  static final int NANOSECONDS_PER_SECONDS = 1_000_000_000;
+  static final int NANOSECONDS_UNIT_UPDATE_THRESHOLD = NANOSECONDS_PER_SECONDS / 2;
```

Comment:
I'm wondering if `UNIT_UPDATE_THRESHOLD_IN_NANOSECONDS` is not better name for this const. Not crucial, yet `ns` thing is least important here so it should not be first word in the constant name (IMHO). Leaving final decision to you though.

Reply:
Done